### PR TITLE
Specviz Spectral Region Getter falls back to reference data units

### LINKED ISSUE #738: [BUG] Specviz: get_spectral_regions fails to apply correct units
This line in get_spectral_regions:

`unit = reg.meta.get("spectral_axis_unit", u.Unit("Angstrom"))`

is inadequate for applying the correct units to the retrieved subsets, since as far as I can tell none of the subsets have that meta attribute populated. We need to update to use a method that actually works to apply the correct units instead of defaulting to Angstrom.

It may be worth working on this and #356 at the same time, although the latter will likely involve fixes deeper in the Glue stack.

[🐱](https://jira.stsci.edu/browse/JDAT-1630)

### PR DESCRIPTION
**Pull request by duytnguyendtn**
### Description

This PR addresses #738 by:
1. Falling back on the reference data's spectral_axis units or
~2. If it can't, defaults to `Counts` as the astropy unit of choice, as opposed to Angstroms.~ Found a bug for this one. Will need to address https://github.com/astropy/specutils/issues/846 and possibly https://github.com/astropy/specutils/issues/847 first. 

I would really appreciate feedback on (2), specifically. I recall us having a conversation about whether Counts are a good fallback unit or whether they actually have a very specific meaning (e.g. like in photo-multiplier tubes :/)

Fixes #738 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by pllim — COMMENTED** [review]
Can we add some tests? 😸

File: jdaviz/configs/specviz/helper.py
Comment (pllim) [inline jdaviz/configs/specviz/helper.py:93]: Counts usually means uncalibrated flux (of some sort). If this is spectral X-axis, maybe pixel is more appropriate... or simple unitless if pixel is misleading.

**Comment by duytnguyendtn** [comment]
@pllim Yup! Probably should have opened this in draft; I wanted to get this out for reviews ASAP since this is needed for monday

File: jdaviz/configs/specviz/helper.py
Comment (duytnguyendtn) [reply jdaviz/configs/specviz/helper.py:93]: Yea @camipacifici corrected me in slack. I think I must have hit my head when I put in that except statement. Cami is recruiting @eteq @PatrickOgle @orifox for help deciding on what to fallback to

**Comment by pllim** [comment]
All of the units that `astropy` supports are listed at https://docs.astropy.org/en/latest/units/index.html#reference-api -- Take your pick! 😸

File: jdaviz/configs/specviz/helper.py
Comment (camipacifici) [reply jdaviz/configs/specviz/helper.py:93]: I would also vote for pixels, but I like the unitless idea if it does not break other things along the way.

File: jdaviz/configs/specviz/helper.py
Comment (pllim) [inline jdaviz/configs/specviz/helper.py:93]: I think this is what the POs decided on from Slack discussions.

```suggestion
                unit = u.dimensionless_unscaled
```

I still recommend adding tests to ensure this does not result in undesirable side effects.

Comment (duytnguyendtn) [reply jdaviz/configs/specviz/helper.py]: Agreed. I pushed in the latest commit!

**Comment by pllim** [comment]
So is this PR blocked by astropy/specutils#846 or not?

**Comment by duytnguyendtn** [comment]
@pllim No, this pixel bug seems to be unrelated to this PR, rather it was exposed during testing. We're going to try to push this forward since I removed that fallback that would have triggered this issue, then fix this pixel-space support in a separate effort

**Comment by pllim** [comment]
Good to hear! I still think a test is necessary regardless.

**Comment by duytnguyendtn** [comment]
Yes! Definitely working on that before this gets merged

**Review by pllim — APPROVED** [review]
Seems fine code wise and look at that green codecov report! 👏 

@camipacifici should still double check that this solves her use case before merge but LGTM.

**Comment by duytnguyendtn** [comment]
Yay for tests! 🎊

**Review by camipacifici — APPROVED** [review]
Looks good to me!

**Pull request merged**
